src/Transport/Utils/AsyncLazy.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus.AzureServiceBus.Utils
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+
+    class AsyncLazy<T> : Lazy<Task<T>>
```

Comment:
Perhaps `LazyAsync`? since it's an extension of Lazy to support task based values as well.